programs/mango-v4/src/instructions/token_liq_bankruptcy.rs

```diff
@@ -59,7 +59,18 @@ pub fn token_liq_bankruptcy(
     let liab_borrow_index = liab_bank.borrow_index;
     let (liqee_liab, liqee_raw_token_index) = liqee.token_position_mut(liab_token_index)?;
     let initial_liab_native = liqee_liab.native(liab_bank);
-    let mut remaining_liab_loss = -initial_liab_native;
+
+    let liqee_health_token_balances =
+        liqee_health_cache.effective_token_balances(HealthType::LiquidationEnd);
+    let liqee_liab_health_balance = liqee_health_token_balances
+        [liqee_health_cache.token_info_index(liab_token_index)?]
+    .spot_and_perp;
+
+    // Allow token bankruptcy only while the spot position and health token position are both negative.
+    // In particular, a very negative perp hupnl does not allow token bankruptcy to happen,
+    // and if the perp hupnl is positive, we need to liquidate that before dealing with token
+    // bankruptcy!
+    let mut remaining_liab_loss = (-initial_liab_native).min(-liqee_liab_health_balance);
```

Comment:
hmm in case of a very negative hupnl, -liqee_liab_health_balance > -initial_liab_native, that means ix would still proceed, contrary to the comment

Reply:
If liab spot or liab balance is positive, the require() will be hit.

If both are negative and liab balance is more negative, then remaining_liab_loss will be -liab_spot. So we only liquidate the actual spot borrows.

Would it be helpful to write `-(initial_liab_native.max(liqee_liab_health_balance))` instead?